subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/release/tasks/UploadGists.java

```diff
@@ -0,0 +1,30 @@
+package org.shipkit.internal.gradle.release.tasks;
+
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.shipkit.internal.notes.util.IOUtil;
+import org.shipkit.internal.util.GitHubApi;
+
+import java.io.File;
+
+public class UploadGists {
+
+    private static final Logger LOG = Logging.getLogger(UploadGistsTask.class);
+
+    public void uploadGists(UploadGistsTask uploadGistsTask) {
+        GitHubApi gitHubApi = new GitHubApi(uploadGistsTask.getGitHubApiUrl(), uploadGistsTask.getGitHubWriteToken());
+        uploadGists(uploadGistsTask, new GistsApi(gitHubApi));
+    }
+
+    public void uploadGists(UploadGistsTask uploadGistsTask, GistsApi gistsApi) {
+        for (File file : uploadGistsTask.getFilesToUpload()) {
+            try {
+                String url = gistsApi.uploadFile(file.getName(), IOUtil.readFully(file));
+                LOG.lifecycle("Gist for file '{}' created. You can find it here: {}", file.getAbsolutePath(), url);
+            } catch (Exception e) {
+                LOG.error("Creating a Gist for '" + file.getAbsolutePath() + "' failed.", e);
```

Comment:
It's not a great UX if we log error with stack trace but we don't really fail the task. Think about it: the users sees "BUILD SUCCESSFUL" and a giant stack trace in the build log. "Error" means failure. If it is not a failure, we can log it as standard message: "Unable to create gist, ignoring. Run with --debug to see stack trace".

For the time being, I suggest to just throw an exception. We can think about the best UX for the failure scenarios.

Reply:
Thanks, right, it doesn't make sense to pass the build if the task fails. 

I changed it a little. Since there may be a lot of these files uploaded, it doesn't make sense to not upload all of them if the first one fails. This is an API call so it can fail. But if any of them throws an exception then I'm throwing one at the end of the task. I will merge it for now to see how it works in practice and if we have a better idea for this error handling then I will improve it later.